src/librvnpal/src/posix/geterrorstring.c

```diff
@@ -1,15 +1,22 @@
-#if defined(__unix__) && !defined(__APPLE__)
-
+#ifndef _GNU_SOURCE
 #define _GNU_SOURCE
+#endif
+
 #include <unistd.h>
 #include <errno.h>
 #include <string.h>
 #include <stdlib.h>
 
 #include "rvn.h"
 #include "status_codes.h"
-
-int32_t rvn_get_error_string(int32_t error, char* buf, int32_t buf_size, int32_t* special_errno_flags) {
+#include "internal_posix.h"
+
+EXPORT int32_t
+rvn_get_error_string(int32_t error, 
```

Comment:
Just one error isn't enough, you need both the special error and the return code.